From patchwork Wed Jul  2 04:01:00 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Vicki Pfau <vi@endrift.com>
X-Patchwork-Id: 14139990
Received: from endrift.com (endrift.com [173.255.198.10])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 76329184
	for <linux-input@vger.kernel.org>; Wed,  2 Jul 2025 04:02:15 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=173.255.198.10
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1751428937; cv=none;
 b=X7LTA6Y0wSOjo0661NDOsCP+4CAJat8QZvbTQd445sJYxfkMdM0ZxpnmvT+ysuwvhUe/NOnZExW9k6MqQlFeyecRHwAS1ohsLh1zzP3xeGj0y+hcW+HOjHUiyB0v5syTPQ5lIFMLJyN5z/bqRQdp1n4Jdw00iKfevXLR0V449Ws=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1751428937; c=relaxed/simple;
	bh=89hBmHmN2y2jy3z0bfOjTaG+MRe06c/jDA+5T6LPZCc=;
	h=From:To:Cc:Subject:Date:Message-ID:In-Reply-To:References:
	 MIME-Version;
 b=G8mapl83pFSbNjWCCOBoUZ3CMJbwPh6+re/EeTQzixoBHbmJhmNMsfiotTkudrtyKjADOeOhQbHVufW3ueY+l+efOfYls/0zm1vzC2d3i8xBNu6nh//UjgIN4NHQR4+sjJ8Pz+uYznsQufqO6ChshfFaCCrden5Ve/r+DTAtMjU=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=endrift.com;
 spf=pass smtp.mailfrom=endrift.com;
 dkim=pass (2048-bit key) header.d=endrift.com header.i=@endrift.com
 header.b=UBhoWzTX; arc=none smtp.client-ip=173.255.198.10
Authentication-Results: smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=endrift.com
Authentication-Results: smtp.subspace.kernel.org;
 spf=pass smtp.mailfrom=endrift.com
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=endrift.com header.i=@endrift.com
 header.b="UBhoWzTX"
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=endrift.com; s=2020;
	t=1751428934; bh=89hBmHmN2y2jy3z0bfOjTaG+MRe06c/jDA+5T6LPZCc=;
	h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
	b=UBhoWzTXogl/MFv+Z8OBg6ql0j2xspO/c6V3a9vL6cg0Py/k9NVz62yjUS6vjwICQ
	 0G/aEQh99+w/6wOPb4PuAzlmerrrZ6+7q7Qf3XdcqPPf/d9C7+9bIDVNYpOFwfa7aR
	 G2dW1IhbBv3ZSBoevC87zfFRvUUGof4fleun4uyK3oIC7s1KP96328FEVQBw3u74b2
	 7KRcawL8EKyCdrpM8s8CRyhb054Lw+3oOB9T367rS5lk5js+bammtLpkCkt8uDbNek
	 TWQ+TRXxr2rzIzTLAT3epHZk763ooF/O0ew4ST1+rzOJ06g9qZdrFXtpR7fAXpidno
	 Hh/0wx5+me8Ag==
Received: from microtis.vulpes.eutheria.net (71-212-74-234.tukw.qwest.net
 [71.212.74.234])
	by endrift.com (Postfix) with ESMTPSA id 798D3A0EA;
	Tue,  1 Jul 2025 21:02:14 -0700 (PDT)
From: Vicki Pfau <vi@endrift.com>
To: Dmitry Torokhov <dmitry.torokhov@gmail.com>,
	Jiri Kosina <jikos@kernel.org>,
	Benjamin Tissoires <bentiss@kernel.org>,
	linux-input@vger.kernel.org
Cc: Vicki Pfau <vi@endrift.com>
Subject: [PATCH 1/3] Input: Add and document BTN_GRIP*
Date: Tue,  1 Jul 2025 21:01:00 -0700
Message-ID: <20250702040102.125432-2-vi@endrift.com>
X-Mailer: git-send-email 2.49.0
In-Reply-To: <20250702040102.125432-1-vi@endrift.com>
References: <20250702040102.125432-1-vi@endrift.com>
Precedence: bulk
X-Mailing-List: linux-input@vger.kernel.org
List-Id: <linux-input.vger.kernel.org>
List-Subscribe: <mailto:linux-input+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-input+unsubscribe@vger.kernel.org>
MIME-Version: 1.0

Many controllers these days have started including grip buttons. As there has
been no particular assigned BTN_* constants for these, they've been
hapharzardly assigned to BTN_TRIGGER_HAPPY*. Unfortunately, the assignemnt of
these has varied significantly between drivers. This patch adds and documents
new constants for these grip buttons.

Signed-off-by: Vicki Pfau <vi@endrift.com>
---
 Documentation/input/gamepad.rst        | 13 +++++++++++++
 drivers/hid/hid-debug.c                |  2 ++
 include/uapi/linux/input-event-codes.h |  5 +++++
 3 files changed, 20 insertions(+)

diff --git a/Documentation/input/gamepad.rst b/Documentation/input/gamepad.rst
index eca17a7f5258..357c138eec66 100644
--- a/Documentation/input/gamepad.rst
+++ b/Documentation/input/gamepad.rst
@@ -190,6 +190,19 @@ Gamepads report the following events:
 
   Rumble is advertised as FF_RUMBLE.
 
+- Grip buttons:
+
+  Many pads include buttons on the rear, usually referred to as either grip or
+  rear buttons, or paddles. These are often reprogrammable by the firmware to
+  appear as "normal" buttons, but are sometimes exposed to software too. Some
+  notable examples of this are the Steam Deck, which has R4, R5, L4, and L5 on
+  the back; the Xbox Elite pads, which have P1-P4; and the Switch 2 Pro
+  Controller, which has GL and GR.
+
+  For these controllers, BTN_GRIPR and BTN_GRIPR2 should be used for the top
+  and bottom (if present) right grip button(s), and BTN_GRIPL and BTN_GRIPL2
+  should be used for the top and bottom (if present) left grip button(s).
+
 - Profile:
 
   Some pads provide a multi-value profile selection switch.  An example is the
diff --git a/drivers/hid/hid-debug.c b/drivers/hid/hid-debug.c
index c6b6b1029540..b25dc3425da3 100644
--- a/drivers/hid/hid-debug.c
+++ b/drivers/hid/hid-debug.c
@@ -3291,6 +3291,8 @@ static const char *keys[KEY_MAX + 1] = {
 	[BTN_TR2] = "BtnTR2",			[BTN_SELECT] = "BtnSelect",
 	[BTN_START] = "BtnStart",		[BTN_MODE] = "BtnMode",
 	[BTN_THUMBL] = "BtnThumbL",		[BTN_THUMBR] = "BtnThumbR",
+	[BTN_GRIPL] = "BtnGripL",		[BTN_GRIPR] = "BtnGripR",
+	[BTN_GRIPL2] = "BtnGripL2",		[BTN_GRIPR2] = "BtnGripR2",
 	[BTN_TOOL_PEN] = "ToolPen",		[BTN_TOOL_RUBBER] = "ToolRubber",
 	[BTN_TOOL_BRUSH] = "ToolBrush",		[BTN_TOOL_PENCIL] = "ToolPencil",
 	[BTN_TOOL_AIRBRUSH] = "ToolAirbrush",	[BTN_TOOL_FINGER] = "ToolFinger",
diff --git a/include/uapi/linux/input-event-codes.h b/include/uapi/linux/input-event-codes.h
index 3b2524e4b667..08cb157ab593 100644
--- a/include/uapi/linux/input-event-codes.h
+++ b/include/uapi/linux/input-event-codes.h
@@ -601,6 +601,11 @@
 #define BTN_DPAD_LEFT		0x222
 #define BTN_DPAD_RIGHT		0x223
 
+#define BTN_GRIPL		0x224
+#define BTN_GRIPR		0x225
+#define BTN_GRIPL2		0x226
+#define BTN_GRIPR2		0x227
+
 #define KEY_ALS_TOGGLE		0x230	/* Ambient light sensor */
 #define KEY_ROTATE_LOCK_TOGGLE	0x231	/* Display rotation lock */
 #define KEY_REFRESH_RATE_TOGGLE	0x232	/* Display refresh rate toggle */

From patchwork Wed Jul  2 04:01:01 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Vicki Pfau <vi@endrift.com>
X-Patchwork-Id: 14139991
Received: from endrift.com (endrift.com [173.255.198.10])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id BF5E01DDA0C
	for <linux-input@vger.kernel.org>; Wed,  2 Jul 2025 04:02:15 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=173.255.198.10
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1751428937; cv=none;
 b=QmB5wwXgH73tH0Q3Znt0US2p34vO9YyebW9krqcAYB7pN4aIWpaA8AYCYSlX+kmWvszGMvlNQuQolgK54fP8qMzOkGFCTlL1EvP8K7kcqn+JboJbcfkw0Y7UjcD5rB1C/VNBlNtyk/WhGWMfsEQeEZalEUleAZoKYULZx2xoK50=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1751428937; c=relaxed/simple;
	bh=U9SETKxmQuNwr6ECFZh0GCHyM6bfoesk6AcanQ1CXrA=;
	h=From:To:Cc:Subject:Date:Message-ID:In-Reply-To:References:
	 MIME-Version;
 b=K0CRijhj9echkfR5ZcciJF8sJBMn0iKCa6lUufKRQYe4H2J5WN1YHcPxOxF03Sf5C/J33yCKvNgI8Otv1naGEgS9Lda0FgrCAG5aTRFRHO6SYwWpGDGtUXP2mBobDGjJxysXg1RnOl617t40dQNzxlMSF+RjibubN4SwIjj8AM8=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=endrift.com;
 spf=pass smtp.mailfrom=endrift.com;
 dkim=pass (2048-bit key) header.d=endrift.com header.i=@endrift.com
 header.b=HczFCmOi; arc=none smtp.client-ip=173.255.198.10
Authentication-Results: smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=endrift.com
Authentication-Results: smtp.subspace.kernel.org;
 spf=pass smtp.mailfrom=endrift.com
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=endrift.com header.i=@endrift.com
 header.b="HczFCmOi"
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=endrift.com; s=2020;
	t=1751428935; bh=U9SETKxmQuNwr6ECFZh0GCHyM6bfoesk6AcanQ1CXrA=;
	h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
	b=HczFCmOiu2BE0w7AgYK6ve0qhqWLIVbmVE1k5X/biPFUG+l2cs7ON+mEukZU2dm5D
	 IBvJ/aADmYtTsInrtiYvLwBAwHee6EgaP81ZNNI3ZqShg4YokJreZJewadATPUFTdf
	 RqyNyVtppzrOGZD8Ac6iHC+RXYhwBYlxU2R4imPUYcyos1BSNlfhO9ErV94SGHFFGJ
	 DnHlXBy1vB7GVYLOt0LsqKPZ70c9sWMk2T3y5pvB/XZWtIZej2SMBTWs1LDtJjkSqh
	 YftbbpyPoSXd86P9kVtwL1rsXXhP6kpOGyleplhzjrvDDVPamr1+kV8xz1oxZTur4n
	 DzNUJiugg1rag==
Received: from microtis.vulpes.eutheria.net (71-212-74-234.tukw.qwest.net
 [71.212.74.234])
	by endrift.com (Postfix) with ESMTPSA id DEB49A124;
	Tue,  1 Jul 2025 21:02:14 -0700 (PDT)
From: Vicki Pfau <vi@endrift.com>
To: Dmitry Torokhov <dmitry.torokhov@gmail.com>,
	Jiri Kosina <jikos@kernel.org>,
	Benjamin Tissoires <bentiss@kernel.org>,
	linux-input@vger.kernel.org
Cc: Vicki Pfau <vi@endrift.com>
Subject: [PATCH 2/3] Input - xpad: Use new BTN_GRIP* buttons
Date: Tue,  1 Jul 2025 21:01:01 -0700
Message-ID: <20250702040102.125432-3-vi@endrift.com>
X-Mailer: git-send-email 2.49.0
In-Reply-To: <20250702040102.125432-1-vi@endrift.com>
References: <20250702040102.125432-1-vi@endrift.com>
Precedence: bulk
X-Mailing-List: linux-input@vger.kernel.org
List-Id: <linux-input.vger.kernel.org>
List-Subscribe: <mailto:linux-input+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-input+unsubscribe@vger.kernel.org>
MIME-Version: 1.0

Signed-off-by: Vicki Pfau <vi@endrift.com>
---
 drivers/input/joystick/xpad.c | 36 +++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/drivers/input/joystick/xpad.c b/drivers/input/joystick/xpad.c
index d02ddfbc5216..90855d54fcc2 100644
--- a/drivers/input/joystick/xpad.c
+++ b/drivers/input/joystick/xpad.c
@@ -478,8 +478,8 @@ static const signed short xpad_abs_triggers[] = {
 
 /* used when the controller has extra paddle buttons */
 static const signed short xpad_btn_paddles[] = {
-	BTN_TRIGGER_HAPPY5, BTN_TRIGGER_HAPPY6, /* paddle upper right, lower right */
-	BTN_TRIGGER_HAPPY7, BTN_TRIGGER_HAPPY8, /* paddle upper left, lower left */
+	BTN_GRIPR, BTN_GRIPR2, /* paddle upper right, lower right */
+	BTN_GRIPL, BTN_GRIPL2, /* paddle upper left, lower left */
 	-1						/* terminating entry */
 };
 
@@ -1073,10 +1073,10 @@ static void xpadone_process_packet(struct usb_xpad *xpad, u16 cmd, unsigned char
 				data[18] = 0;
 
 			/* Elite Series 2 split packet paddle bits */
-			input_report_key(dev, BTN_TRIGGER_HAPPY5, data[18] & BIT(0));
-			input_report_key(dev, BTN_TRIGGER_HAPPY6, data[18] & BIT(1));
-			input_report_key(dev, BTN_TRIGGER_HAPPY7, data[18] & BIT(2));
-			input_report_key(dev, BTN_TRIGGER_HAPPY8, data[18] & BIT(3));
+			input_report_key(dev, BTN_GRIPR, data[18] & BIT(0));
+			input_report_key(dev, BTN_GRIPR2, data[18] & BIT(1));
+			input_report_key(dev, BTN_GRIPL, data[18] & BIT(2));
+			input_report_key(dev, BTN_GRIPL2, data[18] & BIT(3));
 
 			do_sync = true;
 		}
@@ -1173,10 +1173,10 @@ static void xpadone_process_packet(struct usb_xpad *xpad, u16 cmd, unsigned char
 					data[32] = 0;
 
 				/* OG Elite Series Controller paddle bits */
-				input_report_key(dev, BTN_TRIGGER_HAPPY5, data[32] & BIT(1));
-				input_report_key(dev, BTN_TRIGGER_HAPPY6, data[32] & BIT(3));
-				input_report_key(dev, BTN_TRIGGER_HAPPY7, data[32] & BIT(0));
-				input_report_key(dev, BTN_TRIGGER_HAPPY8, data[32] & BIT(2));
+				input_report_key(dev, BTN_GRIPR, data[32] & BIT(1));
+				input_report_key(dev, BTN_GRIPR2, data[32] & BIT(3));
+				input_report_key(dev, BTN_GRIPL, data[32] & BIT(0));
+				input_report_key(dev, BTN_GRIPL2, data[32] & BIT(2));
 			} else if (xpad->packet_type == PKT_XBE2_FW_OLD) {
 				/* Mute paddles if controller has a custom mapping applied.
 				 * Checked by comparing the current mapping
@@ -1186,10 +1186,10 @@ static void xpadone_process_packet(struct usb_xpad *xpad, u16 cmd, unsigned char
 					data[18] = 0;
 
 				/* Elite Series 2 4.x firmware paddle bits */
-				input_report_key(dev, BTN_TRIGGER_HAPPY5, data[18] & BIT(0));
-				input_report_key(dev, BTN_TRIGGER_HAPPY6, data[18] & BIT(1));
-				input_report_key(dev, BTN_TRIGGER_HAPPY7, data[18] & BIT(2));
-				input_report_key(dev, BTN_TRIGGER_HAPPY8, data[18] & BIT(3));
+				input_report_key(dev, BTN_GRIPR, data[18] & BIT(0));
+				input_report_key(dev, BTN_GRIPR2, data[18] & BIT(1));
+				input_report_key(dev, BTN_GRIPL, data[18] & BIT(2));
+				input_report_key(dev, BTN_GRIPL2, data[18] & BIT(3));
 			} else if (xpad->packet_type == PKT_XBE2_FW_5_EARLY) {
 				/* Mute paddles if controller has a custom mapping applied.
 				 * Checked by comparing the current mapping
@@ -1201,10 +1201,10 @@ static void xpadone_process_packet(struct usb_xpad *xpad, u16 cmd, unsigned char
 				/* Elite Series 2 5.x firmware paddle bits
 				 * (before the packet was split)
 				 */
-				input_report_key(dev, BTN_TRIGGER_HAPPY5, data[22] & BIT(0));
-				input_report_key(dev, BTN_TRIGGER_HAPPY6, data[22] & BIT(1));
-				input_report_key(dev, BTN_TRIGGER_HAPPY7, data[22] & BIT(2));
-				input_report_key(dev, BTN_TRIGGER_HAPPY8, data[22] & BIT(3));
+				input_report_key(dev, BTN_GRIPR, data[22] & BIT(0));
+				input_report_key(dev, BTN_GRIPR2, data[22] & BIT(1));
+				input_report_key(dev, BTN_GRIPL, data[22] & BIT(2));
+				input_report_key(dev, BTN_GRIPL2, data[22] & BIT(3));
 			}
 		}
 

From patchwork Wed Jul  2 04:01:02 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Vicki Pfau <vi@endrift.com>
X-Patchwork-Id: 14139992
X-Patchwork-Delegate: jikos@jikos.cz
Received: from endrift.com (endrift.com [173.255.198.10])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 42A7813635E
	for <linux-input@vger.kernel.org>; Wed,  2 Jul 2025 04:02:16 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=173.255.198.10
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1751428937; cv=none;
 b=LJrhp/XIS/wVxvhiyYBdKAeeqfNOEpaYnTGpiNXkvfcFGN+urlJjgr74U6bnGvC5dTCOzXl0xqnEfq9Tr9lJA7AtZHVBZcsxmA9UFN0CCviXdoHWUl4cMI+2fGr1mcHVrHh9chm3ieKOcxFEf9/6wd4drTDxrvSP7P0nZli4IL4=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1751428937; c=relaxed/simple;
	bh=IIAbdE1pYyrf5c8qMI1ZBM9Jwcc2qcMWslWOX7IZI6I=;
	h=From:To:Cc:Subject:Date:Message-ID:In-Reply-To:References:
	 MIME-Version;
 b=UebopsgfQTUsqsczyDKOKCe3XXf4UdCaSojnsJx6D/F4Zl19RA6WBprcy8ombQzRjrIMy7AMZDXuMmDR90BnxB8N1yEBYULLKoKo5T+MEWMwez0GBQc76IX6LOvuO8FRJoO9yYuzkn3QFMe0ti/sJ8kviCLSqFC8C5TcA8ugQUE=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=endrift.com;
 spf=pass smtp.mailfrom=endrift.com;
 dkim=pass (2048-bit key) header.d=endrift.com header.i=@endrift.com
 header.b=glqbcToA; arc=none smtp.client-ip=173.255.198.10
Authentication-Results: smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=endrift.com
Authentication-Results: smtp.subspace.kernel.org;
 spf=pass smtp.mailfrom=endrift.com
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=endrift.com header.i=@endrift.com
 header.b="glqbcToA"
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=endrift.com; s=2020;
	t=1751428935; bh=IIAbdE1pYyrf5c8qMI1ZBM9Jwcc2qcMWslWOX7IZI6I=;
	h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
	b=glqbcToAxLLfvLmZfnL4XSCYZCBeYnahQ/3KzXGHMsq7RaUEzNmkTUurci0234HQL
	 I3EGfkmxyd6cjPCDB1qdDWcsg9D9F2x/rw1fBdpOoiplRPZCP0VYXYGHqytbkaRnGf
	 qrBT4FY8FDdHyBmUmwE4Dvwx27mrZwQder6L/p62bYbV2u74uPYQCKhJvi0oviFFv4
	 MRxsRFVO5423j10ifLFPcD5yzJS7AgFCLadQEtKJJozMKV6280gKutN4hoYsNU8XXP
	 gakjG1hQAZH2nk4KjgpjtYZlzp1flv1ahhtpsEnN4NX+GP96qFczRKNRybzH/9q+tx
	 wV+9o2/g6ZI4Q==
Received: from microtis.vulpes.eutheria.net (71-212-74-234.tukw.qwest.net
 [71.212.74.234])
	by endrift.com (Postfix) with ESMTPSA id 4F4A6A287;
	Tue,  1 Jul 2025 21:02:15 -0700 (PDT)
From: Vicki Pfau <vi@endrift.com>
To: Dmitry Torokhov <dmitry.torokhov@gmail.com>,
	Jiri Kosina <jikos@kernel.org>,
	Benjamin Tissoires <bentiss@kernel.org>,
	linux-input@vger.kernel.org
Cc: Vicki Pfau <vi@endrift.com>
Subject: [PATCH 3/3] HID: hid-steam: Use new BTN_GRIP* buttons
Date: Tue,  1 Jul 2025 21:01:02 -0700
Message-ID: <20250702040102.125432-4-vi@endrift.com>
X-Mailer: git-send-email 2.49.0
In-Reply-To: <20250702040102.125432-1-vi@endrift.com>
References: <20250702040102.125432-1-vi@endrift.com>
Precedence: bulk
X-Mailing-List: linux-input@vger.kernel.org
List-Id: <linux-input.vger.kernel.org>
List-Subscribe: <mailto:linux-input+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-input+unsubscribe@vger.kernel.org>
MIME-Version: 1.0

Signed-off-by: Vicki Pfau <vi@endrift.com>
---
 drivers/hid/hid-steam.c | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/drivers/hid/hid-steam.c b/drivers/hid/hid-steam.c
index 949d307c66a8..dd6595487bef 100644
--- a/drivers/hid/hid-steam.c
+++ b/drivers/hid/hid-steam.c
@@ -757,10 +757,10 @@ static int steam_input_register(struct steam_device *steam)
 	input_set_capability(input, EV_KEY, BTN_THUMB2);
 	if (steam->quirks & STEAM_QUIRK_DECK) {
 		input_set_capability(input, EV_KEY, BTN_BASE);
-		input_set_capability(input, EV_KEY, BTN_TRIGGER_HAPPY1);
-		input_set_capability(input, EV_KEY, BTN_TRIGGER_HAPPY2);
-		input_set_capability(input, EV_KEY, BTN_TRIGGER_HAPPY3);
-		input_set_capability(input, EV_KEY, BTN_TRIGGER_HAPPY4);
+		input_set_capability(input, EV_KEY, BTN_GRIPL);
+		input_set_capability(input, EV_KEY, BTN_GRIPL2);
+		input_set_capability(input, EV_KEY, BTN_GRIPR);
+		input_set_capability(input, EV_KEY, BTN_GRIPR2);
 	} else {
 		input_set_capability(input, EV_KEY, BTN_GEAR_DOWN);
 		input_set_capability(input, EV_KEY, BTN_GEAR_UP);
@@ -1547,8 +1547,8 @@ static void steam_do_input_event(struct steam_device *steam,
  *  9.4  | BTN_SELECT | menu left
  *  9.5  | BTN_MODE   | steam logo
  *  9.6  | BTN_START  | menu right
- *  9.7  | BTN_TRIGGER_HAPPY3 | left bottom grip button
- *  10.0 | BTN_TRIGGER_HAPPY4 | right bottom grip button
+ *  9.7  | BTN_GRIPL2 | left bottom grip button
+ *  10.0 | BTN_GRIPR2 | right bottom grip button
  *  10.1 | BTN_THUMB  | left pad pressed
  *  10.2 | BTN_THUMB2 | right pad pressed
  *  10.3 | --         | left pad touched
@@ -1573,8 +1573,8 @@ static void steam_do_input_event(struct steam_device *steam,
  *  12.6 | --         | unknown
  *  12.7 | --         | unknown
  *  13.0 | --         | unknown
- *  13.1 | BTN_TRIGGER_HAPPY1 | left top grip button
- *  13.2 | BTN_TRIGGER_HAPPY2 | right top grip button
+ *  13.1 | BTN_GRIPL  | left top grip button
+ *  13.2 | BTN_GRIPR  | right top grip button
  *  13.3 | --         | unknown
  *  13.4 | --         | unknown
  *  13.5 | --         | unknown
@@ -1659,8 +1659,8 @@ static void steam_do_deck_input_event(struct steam_device *steam,
 	input_event(input, EV_KEY, BTN_SELECT, !!(b9 & BIT(4)));
 	input_event(input, EV_KEY, BTN_MODE, !!(b9 & BIT(5)));
 	input_event(input, EV_KEY, BTN_START, !!(b9 & BIT(6)));
-	input_event(input, EV_KEY, BTN_TRIGGER_HAPPY3, !!(b9 & BIT(7)));
-	input_event(input, EV_KEY, BTN_TRIGGER_HAPPY4, !!(b10 & BIT(0)));
+	input_event(input, EV_KEY, BTN_GRIPL2, !!(b9 & BIT(7)));
+	input_event(input, EV_KEY, BTN_GRIPR2, !!(b10 & BIT(0)));
 	input_event(input, EV_KEY, BTN_THUMBL, !!(b10 & BIT(6)));
 	input_event(input, EV_KEY, BTN_THUMBR, !!(b11 & BIT(2)));
 	input_event(input, EV_KEY, BTN_DPAD_UP, !!(b9 & BIT(0)));
@@ -1669,8 +1669,8 @@ static void steam_do_deck_input_event(struct steam_device *steam,
 	input_event(input, EV_KEY, BTN_DPAD_DOWN, !!(b9 & BIT(3)));
 	input_event(input, EV_KEY, BTN_THUMB, !!(b10 & BIT(1)));
 	input_event(input, EV_KEY, BTN_THUMB2, !!(b10 & BIT(2)));
-	input_event(input, EV_KEY, BTN_TRIGGER_HAPPY1, !!(b13 & BIT(1)));
-	input_event(input, EV_KEY, BTN_TRIGGER_HAPPY2, !!(b13 & BIT(2)));
+	input_event(input, EV_KEY, BTN_GRIPL, !!(b13 & BIT(1)));
+	input_event(input, EV_KEY, BTN_GRIPR, !!(b13 & BIT(2)));
 	input_event(input, EV_KEY, BTN_BASE, !!(b14 & BIT(2)));
 
 	input_sync(input);
